cmd/fanoutsidecar/main.go

```diff
@@ -99,31 +102,36 @@ func main() {
 		WriteTimeout: writeTimeout,
 	}
 
-	// Start both the manager (which notices ConfigMap changes) and the HTTP server.
-	var g errgroup.Group
-	g.Go(func() error {
-		// set up signals so we handle the first shutdown signal gracefully
-		stopCh := signals.SetupSignalHandler()
-		// Start blocks forever, so run it in a goroutine.
-		return mgr.Start(stopCh)
+	err = mgr.Add(&runnableServer{
+		logger: logger,
+		s:      s,
 	})
-	logger.Info("Fanout sidecar Listening...", zap.String("Address", s.Addr))
-	g.Go(s.ListenAndServe)
-	err = g.Wait()
 	if err != nil {
-		logger.Error("Either the HTTP server or the ConfigMap noticer failed.", zap.Error(err))
+		logger.Fatal("Unable to add ListenAndServe", zap.Error(err))
 	}
 
+	// set up signals so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler()
+	// Start blocks forever.
+	err = mgr.Start(stopCh)
+	if err != nil {
+		logger.Error("manager.Start() returned an error", zap.Error(err))
+	}
+	logger.Info("Exiting...")
```

Comment:
Do you need to do "Exiting..." etc if Start returned an error?

Reply:
I added it because I spent the last two days trying to figure why things weren't working. It turned out that `Start()` returned `nil`. I added the 'Exiting...' message to make it easy to see that `Start()` had returned at all.